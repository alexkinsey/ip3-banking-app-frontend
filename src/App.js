// React and Router
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

// Styles and Themes
import { GlobalStyles } from './common/styles/GlobalStyles';
import { theme } from './common/styles/Theme';
import './App.css';

// Components and Helpers
import { ScrollToTop } from './common/helpers/ScrollToTop';
import { NavBar } from './components/NavBar/NavBar';

// Providers
import { AuthUserProvider } from './providers/AuthUserProvider';
import { CustomerProvider } from './providers/CustomerProvider';
import { AccountsProvider } from './providers/AccountsProvider';

// Routes
import { AppRoutes } from './AppRoutes';

export const App = () => {
  return (
    <ThemeProvider theme={theme}>
      <AuthUserProvider>
        <CustomerProvider>
          <AccountsProvider>
            <GlobalStyles />
            <Router>
              <ScrollToTop />
              <NavBar />
              <AppRoutes />
            </Router>
          </AccountsProvider>
        </CustomerProvider>
      </AuthUserProvider>
    </ThemeProvider>
  );
};
