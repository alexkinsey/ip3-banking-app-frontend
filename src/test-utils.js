import React from 'react';
import { render } from '@testing-library/react';

// Custom hook testing utility
export const renderHookWithContext = (hook, { providerProps, context: Context }) => {
  let hookResult;

  const TestComponent = () => {
    hookResult = hook();
    return null;
  };

  render(
    <Context.Provider {...providerProps}>
      <TestComponent />
    </Context.Provider>
  );

  return hookResult;
};
