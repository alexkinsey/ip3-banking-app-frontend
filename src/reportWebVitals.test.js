import React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils'; 
import { App } from './App';

describe('Index', () => {
  let container;

  beforeEach(() => {
    // Set up a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // Clean up on exiting
    document.body.removeChild(container);
    container = null;
  });

  it('renders the App component correctly', () => {
    // Mock ReactDOM.createRoot to return a mock root
    const createRootMock = jest.spyOn(ReactDOM, 'createRoot').mockReturnValue({
      render: jest.fn(),
      unmount: jest.fn(),
    });

    // Render the component inside an act() callback
    act(() => {
      ReactDOM.createRoot(container).render(<App />);
    });

    // Assert createRoot was called with the container
    expect(createRootMock).toHaveBeenCalledWith(container);

    // Assert the root render function was called with <App />
    expect(createRootMock().render).toHaveBeenCalledWith(<App />);

    // Snapshot test on container innerHTML
    expect(container.innerHTML).toMatchSnapshot();
  });
});
