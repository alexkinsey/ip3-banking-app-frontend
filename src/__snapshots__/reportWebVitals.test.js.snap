// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders the App component correctly 1`] = `""`;
